kate/src/metrics.rs

```diff
@@ -0,0 +1,24 @@
+use crate::BlockDimensions;
+use sp_std::time::Duration;
+
+/// Trait for measurements during the header built process.
+pub trait Metrics {
+	fn extended_block_time(&self, elapsed: Duration);
+	fn preparation_block_time(&self, elapsed: Duration);
+	fn commitment_build_time(&self, elapsed: Duration);
+	fn proof_build_time(&self, elapsed: Duration, cells: u32);
+	fn block_dims_and_size(&self, block_dims: &BlockDimensions, block_len: u32);
+}
+
+/// Adapter to ignore any measurements.
+/// It should be used for testing environments.
+#[derive(Clone, Copy, Debug)]
+pub struct IgnoreMetrics {}
```

Comment:
Should we put this under the `#[cfg(test)]`?

Reply:
As we'd talked, we could find some extra cases (not just testing) for this `struct IgnoreMetrics`.